### PR TITLE
Default docker image uses 64bit JVM

### PR DESCRIPTION
**Pull request by adleong**
* `linkerd:<version>` and `namerd:<version>` now use 64 bit JVMs
* created `linkerd:<version>-32bit` and `namerd:<version>-32bit` images that use the 32 bit JVM
* removed the `minimal` configuration

### CODE REVIEW

File: project/LinkerdBuild.scala [hunk 1]
```diff
@@ -265,31 +265,28 @@ object LinkerdBuild extends Base {
...
+      Iface.controlHttp, Iface.interpreterThrift,
       Namer.consul, Namer.k8s, Namer.marathon, Namer.serversets, Namer.zkLeader,
       Interpreter.perHost, Interpreter.k8s,
       Storage.etcd, Storage.inMemory, Storage.k8s, Storage.zk, Storage.consul
     )
 
+    val LowMemSettings = BundleSettings ++ Seq(
+      dockerJavaImage := "buoyantio/debian-32-bit",
+      dockerTag := s"${version.value}-32bit"
+    )
```
Comment (olix0r) [inline project/LinkerdBuild.scala]: We should ensure that LowMemSettings uses the assemblyJarName defined in appPackagingSettings and not that defined in BundleSettings.

Furthermore, would prefer 32b to 32bit

File: project/LinkerdBuild.scala [hunk 2]
```diff
@@ -318,21 +316,18 @@ object LinkerdBuild extends Base {
     val dcosBootstrap = projectDir("namerd/dcos-bootstrap")
       .dependsOn(core, admin, configCore, Storage.zk)
 
-    val DcosSettings = MinimalSettings ++ Seq(
+    val DcosSettings = BundleSettings ++ Seq(
       assemblyExecScript := dcosExecScript.split("\n").toSeq
     )
```
Comment (olix0r) [inline project/LinkerdBuild.scala]: I think we'll need to fix dockerTag/assemblyJarName/etc for Dcos as well?

File: project/LinkerdBuild.scala [hunk 3]
```diff
@@ -497,45 +489,44 @@ object LinkerdBuild extends Base {
...
       failureAccrual
     )
 
+    val LowmemSettings = BundleSettings ++ Seq(
+      dockerJavaImage := "buoyantio/debian-32-bit",
+      dockerTag := s"${version.value}-32b",
+      assemblyJarName in assembly := s"${name.value}-${version.value}-32b-exec"
+    )
+
+    val LowmemProjects = BundleProjects.filterNot(_ == Project.projectToRef(tls))
```
Comment (olix0r) [inline project/LinkerdBuild.scala]: As far as I can tell, this doesn't really do what we want.  As I mentioned, I think we should just leave all of this in for now.

File: project/LinkerdBuild.scala [hunk 4]
```diff
@@ -497,45 +491,42 @@ object LinkerdBuild extends Base {
...
-      Protocol.h2, Protocol.mux, Protocol.thrift,
+      Protocol.h2, Protocol.http, Protocol.mux, Protocol.thrift,
       Announcer.serversets,
       Telemetry.core, Telemetry.tracelog,
       Tracer.zipkin,
       tls,
       failureAccrual
     )
 
+    val LowmemSettings = BundleSettings ++ Seq(
```
Comment (klingerf) [inline project/LinkerdBuild.scala]: Tiniest of nits: can you capitalize this as `LowMemSettings`, to match the `Namerd.LowMemSettings` val?

File: CHANGES.md [hunk 5]
```diff
@@ -9,6 +9,8 @@
 * Client failure accrual is now configurable via the `failureAccrual` parameter
 * Introduce the grpc-gen and grpc-runtime projects, enabling code
   generation of gRPC clients and servers for Finagle.
+* The default docker images now use a 64 bit JVM.  A `-32b` docker image is
+  also availble but does not support TLS.
```
Comment (olix0r) [inline CHANGES.md]: can we clarify this?

Comment (olix0r) [inline CHANGES.md]: "does not support the boringssl TLS extensions required for ALPN, etc"

**Pull request merged**